pinot-common/src/main/java/com/linkedin/pinot/common/data/DateTimeFieldSpec.java

```diff
@@ -0,0 +1,222 @@
+  }
+
+  // Default constructor required by JSON de-serializer. DO NOT REMOVE.
+  public DateTimeFieldSpec() {
+    super();
+  }
+
+  /**
+   * @param name
+   * @param dataType
+   * @param format - size:timeunit:timeformat eg: 1:MILLISECONDS:EPOCH, 5:MINUTES:EPOCH, 1:DAYS:SIMPLE_DATE_FORMAT:yyyyMMdd
+   * @param granularity - size:timeunit eg: 5:MINUTES
+   */
+  public DateTimeFieldSpec(@Nonnull String name, @Nonnull DataType dataType, @Nonnull String format,
+      @Nonnull String granularity, Object defaultNullValue) {
+    super(name, dataType, true, defaultNullValue);
+    check(name, dataType, format, granularity);
+
+    _format = format;
+    _granularity = granularity;
+  }
+
+
+  /**
+   *
+   * @param name
+   * @param dataType
+   * @param format - size:timeunit:timeformat eg: 1:MILLISECONDS:EPOCH, 5:MINUTES:EPOCH, 1:DAYS:SIMPLE_DATE_FORMAT:yyyyMMdd
+   * @param granularity - size:timeunit eg: 5:MINUTES
+   * @param dateTimeType
+   * @param defaultNullValue
+   */
+  public DateTimeFieldSpec(@Nonnull String name, @Nonnull DataType dataType, @Nonnull String format,
+      @Nonnull String granularity, DateTimeType dateTimeType, Object defaultNullValue) {
+    this(name, dataType, format, granularity, defaultNullValue);
+    _dateTimeType = dateTimeType;
+  }
+
+
+  /**
+   * @param name
+   * @param dataType
+   * @param format - size:timeunit:timeformat eg: 1:MILLISECONDS:EPOCH, 5:MINUTES:EPOCH, 1:DAYS:SIMPLE_DATE_FORMAT:yyyyMMdd
+   * @param granularity - size:timeunit eg: 5:MINUTES
+   */
+  public DateTimeFieldSpec(@Nonnull String name, @Nonnull DataType dataType, @Nonnull String format,
+      @Nonnull String granularity) {
+    super(name, dataType, true);
+    check(name, dataType, format, granularity);
+
+    _format = format;
+    _granularity = granularity;
+  }
+
+
+  /**
+   * @param name
+   * @param dataType
+   * @param format - size:timeunit:timeformat eg: 1:MILLISECONDS:EPOCH, 5:MINUTES:EPOCH, 1:DAYS:SIMPLE_DATE_FORMAT:yyyyMMdd
+   * @param granularity - size:timeunit eg: 5:MINUTES
+   * @param dateTimeType
+   */
+  public DateTimeFieldSpec(@Nonnull String name, @Nonnull DataType dataType, @Nonnull String format,
+      @Nonnull String granularity, DateTimeType dateTimeType) {
+    this(name, dataType, format, granularity);
+    _dateTimeType = dateTimeType;
+  }
+
+  private void check(String name, DataType dataType, String format, String granularity) {
+    Preconditions.checkNotNull(name);
+    Preconditions.checkNotNull(dataType);
+
+    Preconditions.checkNotNull(format);
+    String[] formatTokens = format.split(COLON_SEPARATOR);
+    Preconditions.checkArgument(formatTokens.length == 3 || formatTokens.length == 4,
+        "format must be of format size:timeunit:timeformat(:pattern)");
+    Preconditions.checkArgument(formatTokens[0].matches(NUMBER_REGEX)
+        && EnumUtils.isValidEnum(TimeUnit.class, formatTokens[1]), "format must be of format [0-9]+:<TimeUnit>:<TimeFormat>(:pattern)");
```

Comment:
private static string ERROR_STR = "format must be ...."

Reply:
Done